product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/utils/VFSUtils.java

```diff
@@ -0,0 +1,225 @@
+package org.wso2.carbon.esb.scenario.test.common.utils;
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.commons.net.ftp.FTP;
+import org.apache.commons.net.ftp.FTPClient;
+import org.apache.commons.net.ftp.FTPFile;
+
+public class VFSUtils {
+    private static final Log log = LogFactory.getLog(VFSUtils.class);
+    private static int port = 21;
+
+    /**
+     * Function to upload a file to a FTP Server
+     *
+     * @param hostname host name or IP address of the FTP server
+     * @param userName username of a FTP account on the FTP server to connect to
+     * @param password password corresponds to the username
+     * @param sourceFilePath source file path
+     * @param targetFilePath target path of the remote file
+     * @throws IOException if connecting to FTP server fails or file upload fails or logging out from ftp client fails
+     */
+    public static void ftpFileUpload(String hostname, String userName, String password, String sourceFilePath,
+                                     String targetFilePath) throws IOException {
+        FTPClient ftpClient = connectToFtp(hostname, userName, password);
+
+        try {
+            ftpClient.setFileType(FTP.BINARY_FILE_TYPE);
+            File localFile = new File(sourceFilePath);
+
+            InputStream inputStream = new FileInputStream(localFile);
```

Comment:
We need to close these streams if an error occurs as well. Hence, shall we create these in a try with resources block? Let's do it to all resources that implement the interface 'Closeable"

Reply:
fixed.